shell/platform/tizen/channels/text_input_channel.cc

```diff
@@ -57,6 +57,44 @@ static bool IsASCIIPrintableKey(char c) {
   return false;
 }
 
+static bool TextInputTypeToEcoreIMFInputPanelLayout(
+    std::string text_input_type,
+    Ecore_IMF_Input_Panel_Layout& panel_layout) {
+  bool ret = false;
+  if (text_input_type == "TextInputType.text" ||
+      text_input_type == "TextInputType.multiline") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NORMAL;
+    ret = true;
+  } else if (text_input_type == "TextInputType.number") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_NUMBER;
+    ret = true;
+  } else if (text_input_type == "TextInputType.phone") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PHONENUMBER;
+    ret = true;
+  } else if (text_input_type == "TextInputType.datetime") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_DATETIME;
+    ret = true;
+  } else if (text_input_type == "TextInputType.emailAddress") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_EMAIL;
+    ret = true;
+  } else if (text_input_type == "TextInputType.url") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_URL;
+    ret = true;
+  } else if (text_input_type == "TextInputType.visiblePassword") {
+    panel_layout = ECORE_IMF_INPUT_PANEL_LAYOUT_PASSWORD;
```

Comment:
Note: It seems we need to read the `obscureText` value from `client_config` to fully support password input. Please refer to the Android implementation (`TextInputChannel.java`, `TextInputPlugin.java`) for details.

Reply:
Thanks~ I'll check later